### ISSUE TITLE
Debugger-V2 crashes with KeyError

### ISSUE THREAD

**Issue by dniku** [tensorflow/tensorboard#5482]
### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
Traceback (most recent call last):
  File "<stdin>", line 522, in main
  File "<stdin>", line 75, in wrapper
  File "<stdin>", line 145, in autoidentify
  File "<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/inspect.py", line 1024, in getsource
    lines, lnum = getsourcelines(object)
  File "<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/inspect.py", line 1006, in getsourcelines
    lines, lnum = findsource(object)
  File "<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/inspect.py", line 835, in findsource
    raise OSError('could not get source code')
OSError: could not get source code

--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=9, micro=5, releaselevel='final', serial=0)
INFO: os.name: posix
INFO: os.uname(): posix.uname_result(sysname='Linux', nodename='<redacted>', release='4.19.131-34', version='#1 SMP Fri Jul 10 00:54:12 MSK 2020', machine='x86_64')
INFO: sys.getwindowsversion(): N/A

--- check: package_management
INFO: has conda-meta: True
INFO: $VIRTUAL_ENV: None

--- check: installed_packages
INFO: installed: tensorboard==2.7.0
INFO: installed: tensorflow==2.5.0
INFO: installed: tensorflow-estimator==2.5.0
INFO: installed: tensorboard-data-server==0.6.1

--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.7.0'

--- check: tensorflow_python_version
INFO: tensorflow.__version__: '2.5.0'
INFO: tensorflow.__git_version__: 'v2.5.0-rc3-213-ga4dfb8d1a71'

--- check: tensorboard_data_server_version
INFO: data server binary: '<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/site-packages/tensorboard_data_server/bin/server'
INFO: data server binary version: b'rustboard 0.6.1'

--- check: tensorboard_binary_path
INFO: which tensorboard: b'<redacted>/.miniconda3/envs/dl-tensorflow/bin/tensorboard\n'

--- check: addrinfos
socket.has_ipv6 = True
socket.AF_UNSPEC = <AddressFamily.AF_UNSPEC: 0>
socket.SOCK_STREAM = <SocketKind.SOCK_STREAM: 1>
socket.AI_ADDRCONFIG = <AddressInfo.AI_ADDRCONFIG: 32>
socket.AI_PASSIVE = <AddressInfo.AI_PASSIVE: 1>
Loopback flags: <AddressInfo.AI_ADDRCONFIG: 32>
Loopback infos: [(<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::1', 0, 0, 0)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 0))]
Wildcard flags: <AddressInfo.AI_PASSIVE: 1>
Wildcard infos: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('0.0.0.0', 0)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::', 0, 0, 0))]

--- check: readable_fqdn
INFO: socket.getfqdn(): '<redacted>'

--- check: stat_tensorboardinfo
INFO: directory: /tmp/.tensorboard-info
INFO: os.stat(...): os.stat_result(st_mode=16895, st_ino=203306, st_dev=64521, st_nlink=2, st_uid=31575, st_gid=31575, st_size=4096, st_atime=1639665775, st_mtime=1639665815, st_ctime=1639665815)
INFO: mode: 0o40777

--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/site-packages']; bad_roots (0): []

--- check: full_pip_freeze
INFO: pip freeze --all:
absl-py==0.13.0
argon2-cffi @ file:///home/conda/feedstock_root/build_artifacts/argon2-cffi_1610522573621/work
astunparse==1.6.3
async-generator==1.10
attrs @ file:///home/conda/feedstock_root/build_artifacts/attrs_1620387926260/work
backcall @ file:///home/ktietz/src/ci/backcall_1611930011877/work
bleach @ file:///home/conda/feedstock_root/build_artifacts/bleach_1626371669429/work
cachetools==4.2.2
certifi==2021.5.30
cffi @ file:///home/conda/feedstock_root/build_artifacts/cffi_1625835307225/work
charset-normalizer==2.0.3
cloudpickle==1.6.0
cycler==0.10.0
decorator @ file:///tmp/build/80754af9/decorator_1621259047763/work
defusedxml @ file:///home/conda/feedstock_root/build_artifacts/defusedxml_1615232257335/work
dm-tree==0.1.6
entrypoints @ file:///home/conda/feedstock_root/build_artifacts/entrypoints_1605121927639/work/dist/entrypoints-0.3-py2.py3-none-any.whl
flatbuffers==1.12
gast==0.4.0
google-auth==1.33.0
google-auth-oauthlib==0.4.4
google-pasta==0.2.0
grpcio==1.34.1
h5py==3.1.0
idna==3.2
importlib-metadata @ file:///home/conda/feedstock_root/build_artifacts/importlib-metadata_1625463717820/work
ipykernel @ file:///home/conda/feedstock_root/build_artifacts/ipykernel_1620912942381/work/dist/ipykernel-5.5.5-py3-none-any.whl
ipython @ file:///tmp/build/80754af9/ipython_1617120888991/work
ipython-genutils @ file:///tmp/build/80754af9/ipython_genutils_1606773439826/work
ipywidgets @ file:///home/conda/feedstock_root/build_artifacts/ipywidgets_1609995587151/work
jedi @ file:///tmp/build/80754af9/jedi_1606932531272/work
Jinja2 @ file:///home/conda/feedstock_root/build_artifacts/jinja2_1621419064915/work
jsonschema @ file:///home/conda/feedstock_root/build_artifacts/jsonschema_1614815863336/work
jupyter-client @ file:///tmp/build/80754af9/jupyter_client_1616770841739/work
jupyter-core @ file:///tmp/build/80754af9/jupyter_core_1612213314396/work
jupyterlab-pygments @ file:///home/conda/feedstock_root/build_artifacts/jupyterlab_pygments_1601375948261/work
jupyterlab-widgets @ file:///home/conda/feedstock_root/build_artifacts/jupyterlab_widgets_1609173350931/work
keras-nightly==2.5.0.dev2021032900
Keras-Preprocessing==1.1.2
kiwisolver @ file:///home/conda/feedstock_root/build_artifacts/kiwisolver_1610099772044/work
Markdown==3.3.4
MarkupSafe @ file:///home/conda/feedstock_root/build_artifacts/markupsafe_1621455668600/work
matplotlib @ file:///home/conda/feedstock_root/build_artifacts/matplotlib-suite_1620506038354/work
mistune @ file:///home/conda/feedstock_root/build_artifacts/mistune_1624941317779/work
nbclient @ file:///home/conda/feedstock_root/build_artifacts/nbclient_1614336084111/work
nbconvert @ file:///home/conda/feedstock_root/build_artifacts/nbconvert_1624472721108/work
nbformat @ file:///home/conda/feedstock_root/build_artifacts/nbformat_1617383142101/work
nest-asyncio @ file:///home/conda/feedstock_root/build_artifacts/nest-asyncio_1617163391303/work
notebook @ file:///home/conda/feedstock_root/build_artifacts/notebook_1621259862661/work
numpy @ file:///home/conda/feedstock_root/build_artifacts/numpy_1626648378336/work
oauthlib==3.1.1
olefile @ file:///home/conda/feedstock_root/build_artifacts/olefile_1602866521163/work
opt-einsum==3.3.0
packaging @ file:///home/conda/feedstock_root/build_artifacts/packaging_1625323647219/work
pandas==1.3.0
pandocfilters==1.4.2
parso==0.7.0
patsy==0.5.1
pexpect @ file:///tmp/build/80754af9/pexpect_1605563209008/work
pickleshare @ file:///tmp/build/80754af9/pickleshare_1606932040724/work
Pillow @ file:///tmp/build/80754af9/pillow_1625655408865/work
pip==21.1.3
prometheus-client @ file:///home/conda/feedstock_root/build_artifacts/prometheus_client_1622586138406/work
prompt-toolkit @ file:///tmp/build/80754af9/prompt-toolkit_1616415428029/work
protobuf==3.17.3
ptyprocess @ file:///tmp/build/80754af9/ptyprocess_1609355006118/work/dist/ptyprocess-0.7.0-py2.py3-none-any.whl
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser @ file:///home/conda/feedstock_root/build_artifacts/pycparser_1593275161868/work
pydot @ file:///home/conda/feedstock_root/build_artifacts/pydot_1613571168147/work
Pygments @ file:///tmp/build/80754af9/pygments_1621606182707/work
pyparsing @ file:///home/linux1/recipes/ci/pyparsing_1610983426697/work
PyQt5==5.12.3
PyQt5-sip==4.19.18
PyQtChart==5.12
PyQtWebEngine==5.12.1
pyrsistent @ file:///home/conda/feedstock_root/build_artifacts/pyrsistent_1610146801554/work
python-dateutil @ file:///tmp/build/80754af9/python-dateutil_1626374649649/work
pytz @ file:///home/conda/feedstock_root/build_artifacts/pytz_1612179539967/work
pyzmq==20.0.0
requests==2.26.0
requests-oauthlib==1.3.0
rsa==4.7.2
scipy @ file:///home/conda/feedstock_root/build_artifacts/scipy_1626684351880/work
seaborn @ file:///home/conda/feedstock_root/build_artifacts/seaborn-split_1611834504644/work
Send2Trash @ file:///home/conda/feedstock_root/build_artifacts/send2trash_1624366715919/work
setuptools==52.0.0.post20210125
sip==4.19.13
six @ file:///tmp/build/80754af9/six_1623709665295/work
statsmodels @ file:///home/conda/feedstock_root/build_artifacts/statsmodels_1612273598845/work
tensorboard==2.7.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.0
tensorflow==2.5.0
tensorflow-estimator==2.5.0
tensorflow-probability==0.13.0
termcolor==1.1.0
terminado @ file:///home/conda/feedstock_root/build_artifacts/terminado_1623365326093/work
testpath @ file:///home/conda/feedstock_root/build_artifacts/testpath_1621261527237/work
tornado @ file:///tmp/build/80754af9/tornado_1606942317143/work
tqdm @ file:///home/conda/feedstock_root/build_artifacts/tqdm_1625664838498/work
traitlets @ file:///home/ktietz/src/ci/traitlets_1611929699868/work
typing-extensions==3.7.4.3
urllib3==1.26.6
wcwidth @ file:///tmp/build/80754af9/wcwidth_1593447189090/work
webencodings==0.5.1
Werkzeug==2.0.1
wheel==0.36.2
widgetsnbextension @ file:///home/conda/feedstock_root/build_artifacts/widgetsnbextension_1605475529901/work
wrapt==1.12.1
zipp @ file:///home/conda/feedstock_root/build_artifacts/zipp_1625284368454/work

``````

</details>

## Issue description

I was asked to migrate this issue from https://github.com/tensorflow/tensorflow/issues/53344 .

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution: Docker image based on Ubuntu 18.04 Bionic
- TensorFlow installed from (source or binary): built from patched source code
- TensorFlow version (use command below): `v2.1.3-0-g77f47d6ed6c 2.1.3`
- Python version: `3.6.9`
- Bazel version (if compiling from source): `4.2.1`
- GCC/Compiler version (if compiling from source): `g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0`
- CUDA/cuDNN version:  `libcudnn8/unstable 8.0.5.39-1+cuda11.1 amd64`
- GPU model and memory: Nvidia GeForce RTX 2080

**Describe the current behavior**

I am trying to set up Debugger-V2. My project currently uses Tensorflow 2.1.3 and, as far as I understand, the corresponding Tensorboard version does not support Debugger-V2. Therefore I have set up a separate Conda environment with Tensorboard 2.7.0, where I load logs written using Tensorflow 2.1.3. The "system information" block above provides details on the environment in which the rest of my project runs.

In this setup, the Debugger-V2 UI comes up empty (e.g. it says `Graph Executions (0)` in the top right corner), and there is the following traceback in stdout:

```
> tensorboard --logdir tensorboard_logs/train/ --port 6010
2021-12-07 18:07:08.627387: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0

NOTE: Using experimental fast data loading logic. To disable, pass
    "--load_fast=false" and report issues on GitHub. More details:
    https://github.com/tensorflow/tensorboard/issues/4784

Serving TensorBoard on localhost; to expose to the network, use a proxy or pass --bind_all
TensorBoard 2.7.0 at http://localhost:6010/ (Press CTRL+C to quit)
Exception in thread Thread-5:
Traceback (most recent call last):
  File "<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/threading.py", line 892, in run
    self._target(*self._args, **self._kwargs)
  File "<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/site-packages/tensorboard/plugins/debugger_v2/debug_data_multiplexer.py", line 62, in _run_repeatedly
    target()
  File "<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/site-packages/tensorflow/python/debug/lib/debug_events_reader.py", line 1116, in update
    self._load_graphs()
  File "<redacted>/.miniconda3/envs/dl-tensorflow/lib/python3.9/site-packages/tensorflow/python/debug/lib/debug_events_reader.py", line 1013, in _load_graphs
    debugged_graph = self._graph_by_id[op_creation_proto.graph_id]
KeyError: '0410a4d0-d88b-44f9-b7f5-a49008436b15'
```

The error comes from [this line](https://github.com/tensorflow/tensorflow/blob/d1c4acc743e6f5710e38595ea0c6c876205e8399/tensorflow/python/debug/lib/debug_events_reader.py#L1009).

The directory structure of `tensorboard_logs` is as follows:

```
tensorboard_logs/
└── train
    ├── events.out.tfevents.1638889539.<redacted>.118102.192768.v2
    ├── tfdbg_events.1638889418.<redacted>.execution
    ├── tfdbg_events.1638889418.<redacted>.graph_execution_traces
    ├── tfdbg_events.1638889418.<redacted>.graphs
    ├── tfdbg_events.1638889418.<redacted>.metadata
    ├── tfdbg_events.1638889418.<redacted>.source_files
    └── tfdbg_events.1638889418.<redacted>.stack_frames
```

and I set up debug logging with the following line:

```
        tf.debugging.experimental.enable_dump_debug_info(
            os.path.join(<tensorboard_logs>, 'train'),
            tensor_debug_mode='FULL_TENSOR',  # b/c TF 2.1 doesn't support FULL_HEALTH
            circular_buffer_size=10000)
```

I won't be able to share the log files that cause this crash, and I don't have a minimal reproducing example (yet?). Hopefully this is debuggable without the two.

Aside from the obvious problem with this crash, I am also not sure where I should put debug info (what the first parameter of `enable_dump_debug_info` should be). My logging is done via Keras, and the typical directory structure is as follows:

```
tensorboard_logs/
└── train
└── validation
```

I see 3 places where I could put debug info:

```
tensorboard_logs/  # <- here
└── train  # <- here
└── validation
└── debug_info  # <- here
```

and then I could point Tensorboard to any of those directories. Which is the preferred way?

**Comment by bmd3k** [comment]
Hello @dniku, sorry you are having troubles. 

Although the `enable_dump_debug_info` function was available in TensorFlow 2.1 it was incomplete. You will have to use TensorFlow 2.3 or later in order for it to work properly. Please see the TensorFlow [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md#release-230).

**Comment by bmd3k** [comment]
Your idea to put it in `tensorboard_logs/debug_info` seems reasonable if it works for you.
